lib/jormungandr/test/integration/Test/Integration/Jormungandr/Scenario/API/StakePools.hs

```diff
@@ -128,11 +134,26 @@ spec = do
             expectErrorMessage errMsg405 r
 
     it "STAKE_POOLS_JOIN_01 - Can join a stakepool" $ \ctx -> do
+        w <- fixtureWallet ctx
+
+        -- Join a pool
         (_, p:_) <- eventually $
             unsafeRequest @[ApiStakePool] ctx listStakePoolsEp Empty
-        w <- emptyWallet ctx
-        r <- joinStakePool ctx (p ^. #id) (w, "Secure Passphrase")
-        expectResponseCode HTTP.status403 r
+        joinStakePool ctx (p ^. #id) (w, fixturePassphrase) >>= flip verify
+            [ expectResponseCode HTTP.status200
+            , expectFieldEqual status Pending
+            , expectFieldEqual direction Outgoing
+            ]
+
+        -- Wait for the certificate to be inserted
+        eventually $ do
+            let ep = listTxEp w mempty
+            request @[ApiTransaction n] ctx ep Default Empty >>= flip verify
+                [ expectListItemFieldEqual 0 direction Outgoing
+                , expectListItemFieldEqual 0 status InLedger
+                ]
```

Comment:
👍 

Are we eventually going to test that rewards accumulate in the reward account too?

Reply:
Hm, that must depend on #904. Maybe I should pick it up.

Reply:
yes, requesting joining the stake pool is the first step. the next one is to discover it when accommodated in blockchain